src/uri.ts

```diff
@@ -27,7 +28,7 @@ export const relativeUrl = (from: URL, to: URL): string => RelateUrl.relate(from
 export class URLMap<V> implements Map<URL, V> {
     private map: Map<string, V>
     constructor(entries: Iterable<[URL, V]> = []) {
-        this.map = new Map(map(entries, ([uri, value]): [string, V] => [uri.href, value]))
+        this.map = new Map(of(...entries).pipe(map(([uri, value]): [string, V] => [uri.href, value])))
```

Comment:
I think `from(entries)` would be the better choice here as `...` forces iteration, meaning this iterates the collection twice. `entries` may also potentially be larger than the max argument limit

Reply:
Will update.